### ISSUE TITLE
NullPointerException at DropwizardMetricCollectingClient.defaultMetricName??

### ISSUE THREAD

**Issue by imasahiro** [line/armeria#724]
I just start to investigate the issue. So I'm still not sure whether root cause is located at armeria side.

```
2017-08-15 14:19:19.624  WARN 42997 --- [armeria-common-worker-epoll-3-8] c.l.a.c.l.RequestLogListenerInvoker      : onRequestLog() failed with an exception:
java.lang.NullPointerException: null
    at com.linecorp.armeria.client.metric.DropwizardMetricCollectingClient.defaultMetricName(DropwizardMetricCollectingClient.java:97)
    at com.linecorp.armeria.client.metric.DropwizardMetricCollectingClient.lambda$newDecorator$1(DropwizardMetricCollectingClient.java:88)
    at com.linecorp.armeria.internal.metric.DropwizardMetricCollector.getRequestMetrics(DropwizardMetricCollector.java:101)
    at com.linecorp.armeria.internal.metric.DropwizardMetricCollector.onRequestEnd(DropwizardMetricCollector.java:56)
    at com.linecorp.armeria.common.logging.RequestLogListenerInvoker.invokeOnRequestLog(RequestLogListenerInvoker.java:39)
    at com.linecorp.armeria.common.logging.DefaultRequestLog.notifyListeners(DefaultRequestLog.java:607)
    at com.linecorp.armeria.common.logging.DefaultRequestLog.updateAvailability(DefaultRequestLog.java:565)
    at com.linecorp.armeria.common.logging.DefaultRequestLog.endRequest0(DefaultRequestLog.java:395)
    at com.linecorp.armeria.common.logging.DefaultRequestLog.endRequest(DefaultRequestLog.java:382)
    at com.linecorp.armeria.client.HttpRequestSubscriber.fail(HttpRequestSubscriber.java:258)
    at com.linecorp.armeria.client.HttpRequestSubscriber.failAndRespond(HttpRequestSubscriber.java:268)
    at com.linecorp.armeria.client.HttpRequestSubscriber.writeFirstHeader(HttpRequestSubscriber.java:139)
    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:311)
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
    at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
    at java.lang.Thread.run(Thread.java:745)
```

Here is configuration of decorators.
* version 0.52.0
* Uses armeria-retrofit2

```java
cb.decorator(HttpRequest.class, HttpResponse.class,
             DropwizardMetricCollectingClient.newDecorator(
                     metricRegistry,
                     MetricNames.clientMetricName("MyClient")));
cb.decorator(HttpRequest.class, HttpResponse.class,
             LoggingClient.newDecorator());
cb.decorator(HttpRequest.class, HttpResponse.class, RetryingHttpClient.newDecorator(
        (request, response) -> response.aggregate().handle(
                (message, thrown) -> thrown instanceof ClosedSessionException),
        () -> Backoff.exponential(INITIAL_RETRY_DELAY_MILLIS,
                                  MAX_RETRY_DELAY_MILLIS)
                     .withJitter(0.1)
                     .withMaxAttempts(MAX_ATTEMPTS)));
```

**Comment by trustin** [comment]
Offending line: https://github.com/line/armeria/blob/d07f5d2454406e6a615b318a49d5864e5298d11f/core/src/main/java/com/linecorp/armeria/client/metric/DropwizardMetricCollectingClient.java#L97

It it likely that `HttpHeaders.method()` returned `null`, although I'm not sure why yet. Could you paste/send me what's logged by `LoggingClient`, which will contain the complete headers?

**Comment by trustin** [comment]
Let me add a defensive code to #717 meanwhile.

**Comment by trustin** [comment]
Sort of fixed via #717
